File: Fundamentals_of_Software_Design/1_Basic_skills/Частота вхождения символов/main.c
#include <stdio.h>

void char_frequency(const char *s) {
    int freq[256] = {0}; // Массив для подсчёта символов (ASCII 0-255)

    // Подсчет количества вхождений символов
    for (int i = 0; s[i] != '\0'; i++) {
        freq[(unsigned char)s[i]]++;
    }

    // Вывод результатов
    for (int i = 0; i < 256; i++) {
        if (freq[i] > 0) {
            printf("\"%c\" — %d раз(а).\n", i, freq[i]);
        }
    }
}

int main() {
    char s[] = "aaaabccc";
    char_frequency(s);
    return 0;
}

File: Fundamentals_of_Software_Design/1_Basic_skills/Частота вхождения символов/main.py
def char_frequency(s):
    freq = {}
    for char in s:
        if char in freq:
            freq[char] += 1
        else:
            freq[char] = 1
    return freq


# Пример использования
s = "aaaabccc"
freq = char_frequency(s)

for char, count in freq.items():
    print(f'"{char}" — {count} раз(а).')
